src/merkle_tree.rs

```diff
@@ -0,0 +1,221 @@
+use hmac_sha256::Hash;
+
+pub struct MerkleTree {
+    pub root: Option<[u8; 32]>,
+    pub leafs: Vec<[u8; 32]>,
+}
+
+impl Default for MerkleTree {
+    fn default() -> Self {
+        Self {
+            root: None,
+            leafs: Vec::new(),
+        }
+    }
+}
```

Comment:
I'm not sure a `default` implementation makes sense for `MerkleTree`.

Reply:
Okay! I'm not really sure why I included it originally.

Reply:
I remember now why I added the `Default` trait! It was suggested to me by clippy:

```
% cargo clippy
    Checking merkle-tree v0.1.0 (/Users/luca/Documents/merkle-tree)
warning: you should consider adding a `Default` implementation for `MerkleTree`
  --> src/merkle_tree.rs:10:5
   |
10 | /     pub fn new() -> Self {
11 | |         Self {
12 | |             root: None,
13 | |             leaves: Vec::new(),
14 | |         }
15 | |     }
   | |_____^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#new_without_default
   = note: `#[warn(clippy::new_without_default)]` on by default
help: try adding this
   |
8  + impl Default for MerkleTree {
9  +     fn default() -> Self {
10 +         Self::new()
11 +     }
12 + }
   |

warning: `merkle-tree` (lib) generated 1 warning (run `cargo clippy --fix --lib -p merkle-tree` to apply 1 suggestion)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.14s
```

Reply:
I would say an empty `new` doesn't make much sense either.

Reply:
I see, I think you're right.